### PR TITLE
BUG: Fix #25481 by fixing the error message in TypeError

### LINKED ISSUE #25481: BUG: confusing error message for DataFrame.plot()
see also https://github.com/pandas-dev/pandas/issues/17372#issuecomment-325759975

#### Code Sample, a copy-pastable example if possible

```python
import pandas as pd
df = pd.DataFrame({'A': ["x", "y", "z"], 'B': [1, 2, 3]})
df['A'].plot()
```
#### Problem description

```python-traceback
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-4-06628911498f> in <module>
      1 import pandas as pd
      2 df = pd.DataFrame({'A': ["x", "y", "z"], 'B': [1, 2, 3]})
----> 3 df['A'].plot()

~\Anaconda3\lib\site-packages\pandas\plotting\_core.py in __call__(self, kind, ax, figsize, use_index, title, grid, legend, style, logx, logy, loglog, xticks, yticks, xlim, ylim, rot, fontsize, colormap, table, yerr, xerr, label, secondary_y, **kwds)
   2739                            colormap=colormap, table=table, yerr=yerr,
   2740                            xerr=xerr, label=label, secondary_y=secondary_y,
-> 2741                            **kwds)
   2742     __call__.__doc__ = plot_series.__doc__
   2743 

~\Anaconda3\lib\site-packages\pandas\plotting\_core.py in plot_series(data, kind, ax, figsize, use_index, title, grid, legend, style, logx, logy, loglog, xticks, yticks, xlim, ylim, rot, fontsize, colormap, table, yerr, xerr, label, secondary_y, **kwds)
   2000                  yerr=yerr, xerr=xerr,
   2001                  label=label, secondary_y=secondary_y,
-> 2002                  **kwds)
   2003 
   2004 

~\Anaconda3\lib\site-packages\pandas\plotting\_core.py in _plot(data, x, y, subplots, ax, kind, **kwds)
   1802         plot_obj = klass(data, subplots=subplots, ax=ax, kind=kind, **kwds)
   1803 
-> 1804     plot_obj.generate()
   1805     plot_obj.draw()
   1806     return plot_obj.result

~\Anaconda3\lib\site-packages\pandas\plotting\_core.py in generate(self)
    256     def generate(self):
    257         self._args_adjust()
--> 258         self._compute_plot_data()
    259         self._setup_subplots()
    260         self._make_plot()

~\Anaconda3\lib\site-packages\pandas\plotting\_core.py in _compute_plot_data(self)
    371         if is_empty:
    372             raise TypeError('Empty {0!r}: no numeric data to '
--> 373                             'plot'.format(numeric_data.__class__.__name__))
    374 
    375         self.data = numeric_data

TypeError: Empty 'DataFrame': no numeric data to plot
```

#### Expected Output

```python-traceback
TypeError: no numeric data to plot
```
#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 3.7.1.final.0
python-bits: 64
OS: Windows
OS-release: 10
machine: AMD64
processor: Intel64 Family 6 Model 158 Stepping 9, GenuineIntel
byteorder: little
LC_ALL: None
LANG: None
LOCALE: None.None

pandas: 0.23.4
pytest: 4.0.2
pip: 18.1
setuptools: 40.6.3
Cython: 0.29.2
numpy: 1.15.4
scipy: 1.1.0
pyarrow: None
xarray: None
IPython: 7.2.0
sphinx: 1.8.2
patsy: 0.5.1
dateutil: 2.7.5
pytz: 2018.7
blosc: None
bottleneck: 1.2.1
tables: 3.4.4
numexpr: 2.6.8
feather: None
matplotlib: 3.0.2
openpyxl: 2.5.12
xlrd: 1.2.0
xlwt: 1.3.0
xlsxwriter: 1.1.2
lxml: 4.2.5
bs4: 4.6.3
html5lib: 1.0.1
sqlalchemy: 1.2.15
pymysql: None
psycopg2: None
jinja2: 2.10
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None

</details>

### PR DESCRIPTION
**Pull request by bharatr21**
BUG: Fix #25481 by fixing the error message in TypeError

- [x] closes #25481
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Comment by simonjayhawkins** [comment]
@Bharat123rox : a bug fix requires a whatsnew entry... and maybe more tests.

these tests hit the line of code changed..

```
pandas/tests/plotting/test_datetimelike.py::TestTSPlot::test_nonnumeric_exclude 
pandas/tests/plotting/test_frame.py::TestDataFramePlots::test_subplots_timeseries_y_axis 
pandas/tests/plotting/test_frame.py::TestDataFramePlots::test_all_invalid_plot_data
pandas/tests/plotting/test_frame.py::TestDataFramePlots::test_partially_invalid_plot_data
pandas/tests/plotting/test_series.py::TestSeriesPlots::test_invalid_plot_data
pandas/tests/plotting/test_series.py::TestSeriesPlots::test_partially_invalid_plot_data
```

so it may be possible to just use the match parameter of pytest.raises in these tests.

File: doc/source/whatsnew/v0.25.0.rst [hunk 1]
```diff
@@ -121,7 +121,7 @@ Performance Improvements
 Bug Fixes
 ~~~~~~~~~
 - Bug in :func:`to_datetime` which would raise an (incorrect) ``ValueError`` when called with a date far into the future and the ``format`` argument specified instead of raising ``OutOfBoundsDatetime`` (:issue:`23830`)
--
+- Bug in :func: `plot` for a `pd.Series` or a `pd.DataFrame` which had a confusing error message mentioning an "Empty" DataFrame, now raises ``TypeError: no numeric data to plot`` (:issue:`25481`)
```
Comment (TomAugspurger) [inline doc/source/whatsnew/v0.25.0.rst]: Just refer to it as :meth:\`DataFrame.plot\`.

Probably don't need to include the error message here. Rather, say something like "Improved the error message of ..."

**Comment by jreback** [comment]
lgtm. @TomAugspurger

File: pandas/plotting/_core.py [hunk 2]
```diff
@@ -360,8 +360,7 @@ def _compute_plot_data(self):
 
         # no empty frames or series allowed
```
Comment (WillAyd) [inline pandas/plotting/_core.py]: Not a blocker for me but might be nice to clarify this comment. On initial glance it implies to me that the entire passed Frame might have been empty, whereas it really only y applies where numeric data doesn't exist, which is slightly different

**Comment by jreback** [comment]
thanks @Bharat123rox

**Pull request merged**